File: Client/Cell.cs
using System;
using System.Numerics;
using Raylib_cs;

namespace Client
{
    public class Cell
    {
        // screen size
        protected int screenWidth;
        protected int screenHeight;

        // visualization properties
        protected float posx;
        protected float posy;
        protected int radius = 5;
        protected int updateFactor;
        public Vector2 position;
        protected Color color;

        // constructor
        public Cell(int screenWidth, int screenHeight, int updateFactor, Color color)
        {
            // set screenHeight
            this.screenWidth = screenWidth;
            this.screenHeight = screenHeight;

            this.updateFactor = updateFactor;

            this.color = color;

            posx = (float)screenWidth / 2 + 20;
            posy = (float)screenHeight / 2 + 20;
        }

        //------methods------//

        public Vector2 getPosition()
        {
            return position;
        }

        public void updatePosition()
        {
            // get probs for updating
            Random rnd = new Random();
            int probx = rnd.Next(0, 101);
            int proby = rnd.Next(0, 101);

            if ((posx + updateFactor) <= screenWidth - radius && probx <= 50)
            {
                posx += updateFactor;
            }
            else if (posx - updateFactor >= radius)
            {
                posx -= updateFactor;
            }
            // y up or down
            if ((posy + updateFactor) <= screenHeight - radius && proby <= 50)
            {
                posy += updateFactor;
            }
            else if (posy - updateFactor >= radius)
            {
                posy -= updateFactor;
            }

            position.X = posx;
            position.Y = posy;
        }

        public void draw()
        {
            Raylib.DrawCircleV(position, radius, this.color);
        }

    }
}

File: ConvertToPython/Cell.py
import pyray as pr
import random

class Cell:
    def __init__(self, screen_width: int, screen_height: int,  posx: int, posy: int, radius: int, color):
        self.screen_width = screen_width
        self.screen_height = screen_height
        self.color = color

        self.radius = radius

        self.posx = posx
        self.posy = posy
        
        self.position = (self.posx, self.posy)

    # ---------- methods ----------- #
    def getPosition(self) -> dict:
        return self.position
    
    def updatePosition(self, update_factor_x, update_factor_y) -> None:
        # update randomly for now
        random_num_x = random.randint(0,11) # get num to be threshold
        random_num_y = random.randint(0,11)

        # update self.posx
        if (self.posx + update_factor_x <= self.screen_width - self.radius and random_num_x <= 50):
            self.posx += update_factor_x 
        elif self.posx - update_factor_x >= self.radius:
            self.posx -= update_factor_x

        # update self.posy
        if self.posy + update_factor_y <= self.screen_height - self.radius and random_num_y > 50:
            self.posy += update_factor_y 
        elif self.posy - update_factor_y >= self.radius:
            self.posy -= update_factor_y

        # update position
        self.position = (self.posx, self.posy)
    
    def draw(self):
        pr.draw_circle(self.posx, self.posy, self.radius, self.color)